### PR TITLE
[Improve][Connector-V2-kafka] Support setting read starting offset or time at startup config

### LINKED ISSUE #2959: [Improve][Connector-V2-kafka] Support setting read starting offset or time at startup config
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

subtask of #2755 

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

close #2959

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Please refer to [Coding Guide](https://seatunnel.apache.org/docs/contribution/coding-guide#how-to-submit-a-high-quality-pull-request) and change your pull request titile. Thx.

**Comment by Carl-Zhou-CN** [comment]
> please refer to [Coding Guide](https://seatunnel.apache.org/docs/contribution/coding-guide#how-to-submit-a-high-quality-pull-request) and change your pull request titile. Thx.

ok

**Comment by Carl-Zhou-CN** [comment]
@Hisoka-X @TyrantLucifer   Please help to review

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java [hunk 1]
```diff
@@ -91,20 +122,20 @@ public void close() throws IOException {
...
         }
     }
 
-    private Collection<? extends KafkaSourceSplit> convertToNextSplit(List<KafkaSourceSplit> splits) {
+    private Map<TopicPartition, ? extends KafkaSourceSplit> convertToNextSplit(List<KafkaSourceSplit> splits) {
         try {
-            Map<TopicPartition, ListOffsetsResult.ListOffsetsResultInfo> listOffsets =
-                getKafkaPartitionLatestOffset(splits.stream().map(KafkaSourceSplit::getTopicPartition).collect(Collectors.toList()));
+            Map<TopicPartition, Long> listOffsets =
+                listOffsets(splits.stream().map(KafkaSourceSplit::getTopicPartition).collect(Collectors.toList()), OffsetSpec.earliest());
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java]: Why use `OffsetSpec.earliest()` not latest?

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java]: Sorry, it was my mistake. I should have used latest

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java]: Maybe you should add some test case.

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java]: Ok, I will add e2e tests later

**Review by Hisoka-X — APPROVED** [review]
LGTM

File: docs/en/connector-v2/source/kafka.md [hunk 2]
```diff
@@ -66,6 +69,24 @@ The structure of the data, including field names and field types.
...
+The offset required for consumption mode to be specific_offsets
+for example:
+```hocon
+   start_mode.offsets = {
+            info-0 = 70
+            info-1 = 10
+            info-2 = 10
+         }
+```
+
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/kafka.md:89]: Please add `changed log` reference https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/source/Redis.md#next-version

**Review by EricJoy2048 — APPROVED** [review]
I will merge this pr, and I will create a pr to add change log.

File: docs/en/connector-v2/source/kafka.md (hunk 2 above)
Comment (Carl-Zhou-CN) [reply docs/en/connector-v2/source/kafka.md:89]: ok，thank you

**Pull request merged**